### ISSUE TITLE
Remove `./configure` and require `autoconf` for building Citus

### ISSUE THREAD

**Issue by lithp** [citusdata/citus#1984]
**Proposal**:
- Require that everyone who builds Citus has `autoconf` installed, so they're able to build Citus
- Remove `./configure` from the repository

**Benefits**

This will make diffs smaller and more meaningful when they include changes to `./configure.in`. It also reduces the (admittedly already small) chance of merge conflicts if multiple branches edit `./configure`.

**Rationale**

Different versions of `autoconf` emit different `./configure` files, so we have large diffs whenever `./configure.in` changes, depending on who makes the change.

@byucesoy [made a valiant effort](https://github.com/citusdata/citus/pull/1769) at solving the problem but I think there's an easier solution than trying to convince all autoconf versions to emit the same file.

Some notes:
- This doesn't involve messing with our Makefile. Our Makefile already generates `./configure` when `configure.in` changes, so removing `./configure` is all this change takes.

**Comment by jasonmp85** [comment]
I think I'm OK with this. @anarazel originally set it up, but he understandably not chime in on something this trivial. That said, why not.

**Comment by byucesoy** [comment]
Yep, that makes sense. It is fair requirement if you want to build Citus from source.

**Comment by marcocitus** [comment]
If we remove `configure` we'll break a few build pipelines and make it slightly harder to compile from source for users unfamiliar with `autoconf`. We're averaging about 50 clones a day, so I doubt that it's just our devs building Citus from source.

Postgres has included a `configure` file in the repo since 1997, and it seems to have not become a significant problem.

What if we just have a recommended autoconf version?
